### DISCUSSION TITLE
Allow exporting the SPA source code

### DISCUSSION THREAD

**Discussion by ghost** [Budibase/budibase#1702] [Ideas]
This idea might be far-fetched, but I was wondering if it might be possible to export the source code of the SPA. The reason I wanted this is because I already have a backend API and I wanted to use budibase to quickly design the backend interface. This would allow me to side step having to host budibase myself and deploy the SPA to somewhere like netlify. I would also be able to easily integrate complex logic myself.

**Comment by mjashanks** [comment]
Thanks for this. Exporting the front end is not as easy as it sounds to get something useful. This is not going to happen anytime soon.

Your only option here is to connect Budibase's backend to your backend.

You could do this using the external data sources, using either a REST connector, or directly to your DB.

We are in the process of improving these, and in the coming months you will be able to write your own, with NodeJS, and import.

**Reply by deepakchampatiray** [reply]
Hey @mjashanks, the idea of being able to write our own connectors using NodeJS is indeed a good one. Is there an alpha program to register to? Where we can get a sneak-peak of how it is supposed to be done.

**Reply by mjashanks** [reply]
We haven't started work on it yet, but we will update this discussion when we do.

In the meantime, it would be really helpful if you could give a description of what you would like to be able to do with a custom connector. 

Thanks!

**Reply by deepakchampatiray** [reply]
Thanks @mjashanks for the info. Here is what I'm exploring to do. I work with a proprietary business workflow application called IBM BPM. It is a self contained system ina sense that it provides drag-and-drop UI and integration building capabilities. But the UI builder is complicated and use and difficult to extend. So, many customer end up building external UI applications using React/Angular and communicate with the workflow system using REST api. If I can integrate budibase with the Workflow system, then this problem of building external UI from scratch can be mitigated. Also, data-objects of the workflow can "theoretically" be introspected to auto-generate the UI components. This opens up a huge market for budibase (Many of the >3000 enterprise customers of IBM BPM).

Please drop me a note on deepak@champatiray[dot]com if you think it would be an worth endeavour to pursue.

**Comment by slfhstr** [comment]
Try using GrapesJS for using front-end design
